pkg/smokescreen/smokescreen.go

```diff
@@ -938,6 +939,25 @@ func checkACLsForRequest(config *Config, req *http.Request, destination hostport
 	// which is the behavior we want here.
 	connectProxyHost := req.Header.Get("X-Upstream-Https-Proxy")
 
+	if connectProxyHost != "" {
+		connectProxyUrl, err := url.Parse(connectProxyHost)
+
+		if err != nil {
+			config.Log.WithFields(logrus.Fields{
+				"error":               err,
+				"role":                role,
+				"upstream_proxy_name": req.Header.Get("X-Upstream-Https-Proxy"),
+				"destination_host":    destination.Host,
+				"kind":                "parse_failure",
+			}).Error("Unable to parse X-Upstream-Https-Proxy header.")
+
+			config.MetricsClient.Incr("acl.upstream_proxy_error", 1)
```

Comment:
Could you tag this with the failure kind?

Reply:
Changed `upstream_proxy_error` to `upstream_proxy_parse_error` for clarify 

The logging above already has the "kind" field set - is there something else you were imagining here?